src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -0,0 +1,483 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.remote;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.MutableCapabilities;
+import org.openqa.selenium.ScreenOrientation;
+import org.openqa.selenium.remote.CapabilityType;
+
+import java.net.URL;
+import java.util.List;
+
+public abstract class MobileOptions<T extends MobileOptions<T>> extends MutableCapabilities {
+
+    public String getPlatformName() {
+        return (String) getCapability(CapabilityType.PLATFORM_NAME);
+    }
+
+    /**
+     * Set the url for the location of the App.
+     *
+     * @param url is a String representing the location of the App
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#APP
+     */
+    public T setApp(String url) {
+        return amend(MobileCapabilityType.APP, url);
+    }
+
+    /**
+     * Set the url for the location of the App.
+     *
+     * @param url is the URL representing the location of the App
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#APP
+     */
+    public T setApp(URL url) {
```

Comment:
why only URL? this could also be a path to a local file

Reply:
It supports String as well. https://github.com/appium/java-client/pull/1319/files#diff-66811b841580b850e3df6d3581ea3b83R72